profiler/models/profiler_profile.py

```diff
@@ -0,0 +1,470 @@
+                            self.env['profiler.profile.python.line'].create({
+                                'cprof_tottime': data['tottime'],
+                                'cprof_ncalls': data['calls'],
+                                'cprof_nrcalls': data['rcalls'],
+                                'cprof_ttpercall': data['tt_percall'],
+                                'cprof_cumtime': data['cumtime'],
+                                'cprof_ctpercall': data['ct_percall'],
+                                'cprof_fname': (
+                                    "%(file)s:%(lineno)s (%(method)s)" % data),
+                                'profile_id': self.id,
+                            })
+                        attachment.index_content = py_stats
+                except IOError:
+                    # Fancy feature but not stop process if fails
+                    _logger.info("There was an error while getting the stats"
+                                 "from the cprofile_path")
+                    # pylint: disable=unnecessary-pass
+                    pass
+                self.dump_postgresql_logs()
+                _logger.info("cProfile stats stored.")
+            else:
+                _logger.info("cProfile stats empty.")
+        return attachment
+
+    @api.multi
+    def clear(self, reset_date=True):
+        self.ensure_one()
+        _logger.info("Clear profiler")
+        if reset_date:
+            self.date_started = self.now_utc()
+        ProfilerProfile.profile.clear()
+
+    @api.multi
+    def disable(self):
+        self.ensure_one()
+        _logger.info("Disabling profiler")
+        ProfilerProfile.enabled = False
+        self.state = 'disabled'
+        self.date_finished = self.now_utc()
+        self.dump_stats()
+        self.clear(reset_date=False)
+        self._reset_postgresql()
+
+    @staticmethod
+    @contextmanager
+    def profiling():
+        """Thread local profile management, according to the shared "enabled"
+        """
+        if ProfilerProfile.enabled:
+            _logger.debug("Catching profiling")
+            ProfilerProfile.profile.enable()
+        try:
+            yield
+        finally:
+            if ProfilerProfile.enabled:
+                ProfilerProfile.profile.disable()
+
+    @api.multi
+    def action_view_attachment(self):
+        attachments = self.env['ir.attachment'].search([
+            ('res_model', '=', self._name), ('res_id', '=', self.id)])
+        action = self.env.ref("base.action_attachment").read()[0]
+        action['domain'] = [('id', 'in', attachments.ids)]
+        return action
+
+    @api.model
+    def set_pgoptions_enabled(self):
+        """Verify if postgresql has configured the parameters for logging"""
+        ProfilerProfile.pglogs_enabled = True
+        pgoptions_enabled = bool(os.environ.get('PGOPTIONS'))
+        _logger.info('Logging enabled from environment '
+                     'variable PGOPTIONS? %s', pgoptions_enabled)
```

Comment:
Does this make sense? :thinking: What if Postgres is in a different server?

To check enabled options, shouldn't you get those from the database itself?